misc/dist/.gitignore

```diff
@@ -0,0 +1,2 @@
+# Directory caught by the dependency .gitignore
+!ios_xcode/godot_ios/vulkan/icd.d
```

Comment:
I would just put this in the main `.gitignore` with an absolute path.

Reply:
`vulkan/icd.d` files can be removed, we use static libs for MoltenVK, so these are useless.

Reply:
I'm happy to delete 'em if you want; this feels out of my paygrade, if that makes sense, but if it's the right choice I can just do it.